worker/caasunitprovisioner/application_worker.go

```diff
@@ -189,6 +206,27 @@ func (aw *applicationWorker) loop() error {
 					return errors.Trace(err)
 				}
 			}
+		case _, ok := <-appOperatorWatcher.Changes():
+			if !ok {
+				logger.Debugf("%v", appOperatorWatcher.Wait())
+				worker.Stop(appOperatorWatcher)
+				appOperatorWatcher = nil
+				continue
+			}
+			logger.Debugf("operator update for %v", aw.application)
+			operator, err := aw.containerBroker.Operator(aw.application)
+			if errors.IsNotFound(err) {
+				logger.Debugf("pod not found for application %q", aw.application)
+				if err := aw.provisioningStatusSetter.SetOperatorStatus(aw.application, status.Terminated, "", nil); err != nil {
```

Comment:
This will drop out to the code below and set the status again if err is nil
Probs the easiest fix is to put the last SetOperatorStatus() call below inside an else{}